tools/src/test/java/org/apache/kafka/tools/GetOffsetShellTest.java

```diff
@@ -0,0 +1,160 @@
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@ExtendWith(value = ClusterTestExtensions.class)
+@ClusterTestDefaults(clusterType = Type.ZK)
+@Tag("integration")
+public class GetOffsetShellTest {
+    private final int topicCount = 4;
+    private final int offsetTopicPartitionCount = 4;
+    private final ClusterInstance cluster;
+
+    public GetOffsetShellTest(ClusterInstance cluster) {
+        this.cluster = cluster;
+    }
+
+    private String getTopicName(int i) {
+        return "topic" + i;
+    }
+
+    @BeforeEach
+    public void setUp() {
+    }
+
+    static class Row {
+        private String name;
+        private int partition;
+        private Long timestamp;
+
+        public Row(String name, int partition, Long timestamp) {
+            this.name = name;
+            this.partition = partition;
+            this.timestamp = timestamp;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (o == this) return true;
+
+            if (!(o instanceof Row)) return false;
+
+            Row r = (Row) o;
+
+            return name.equals(r.name) && partition == r.partition && Objects.equals(timestamp, r.timestamp);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(name, partition, timestamp);
+        }
+    }
+
+    @ClusterTest
+    public void testNoFilterOptions() {
+        try(Admin admin = Admin.create(cluster.config().adminClientProperties())) {
+            List<NewTopic> topics = new ArrayList<>();
+
+            IntStream.range(0, topicCount + 1).forEach(i -> topics.add(new NewTopic(getTopicName(i), i, (short)1)));
+
+            admin.createTopics(topics);
+        }
+
+        Properties props = new Properties();
+        props.put(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, cluster.config().producerProperties().get("bootstrap.servers"));
+        props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+        props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+
+        try (KafkaProducer<String, String> producer = new KafkaProducer<>(props)) {
+            IntStream.range(0, topicCount + 1)
+                     .forEach(i -> IntStream.range(0, i * i)
+                            .forEach(msgCount -> producer.send(
+                                    new ProducerRecord<>(getTopicName(i), msgCount % i, null, "val" + msgCount)))
+                     );
+        }
```

Comment:
I decided to fit the tests for that type of test extensions, but now I struggle with environment preparation. I don't know how to get access to `cluser` in `@BeforeEach`. `@ClusterTest` annotation doesn't help, so I'd like to know, how can I create topics before tests.